diff --git a/node_modules/@node-saml/node-saml/node_modules/xml-encryption/lib/xmlenc.js b/node_modules/@node-saml/node-saml/node_modules/xml-encryption/lib/xmlenc.js
index 0ec08dd..78bad0d 100644
--- a/node_modules/@node-saml/node-saml/node_modules/xml-encryption/lib/xmlenc.js
+++ b/node_modules/@node-saml/node-saml/node_modules/xml-encryption/lib/xmlenc.js
@@ -242,7 +242,7 @@ function decryptKeyInfo(doc, options) {
     xpath.select("//*[local-name(.)='CipherValue']", keyInfo)[0];
 
   switch (keyEncryptionAlgorithm) {
-    case 'http://www.w3.org/2001/04/xmlenc#rsa-oaep-mgf1p':
+    case 'http://www.w3.org/2009/xmlenc11#rsa-oaep':
       return decryptKeyInfoWithScheme(encryptedKey, options, 'RSA-OAEP');
     case 'http://www.w3.org/2001/04/xmlenc#rsa-1_5':
       utils.warnInsecureAlgorithm(keyEncryptionAlgorithm, options.warnInsecureAlgorithm);
@@ -255,7 +255,7 @@ function decryptKeyInfo(doc, options) {
 function decryptKeyInfoWithScheme(encryptedKey, options, scheme) {
   var padding = scheme === 'RSA-OAEP' ? crypto.constants.RSA_PKCS1_OAEP_PADDING : crypto.constants.RSA_PKCS1_PADDING;
   var key = Buffer.from(encryptedKey.textContent, 'base64');
-  var decrypted = crypto.privateDecrypt({ key: options.key, padding: padding}, key);
+  var decrypted = crypto.privateDecrypt({ key: options.key, padding: padding, oaepHash: 'sha256'}, key);
   return Buffer.from(decrypted, 'binary');
 }
 
