diff --git a/node_modules/xml-encryption/lib/xmlenc.js b/node_modules/xml-encryption/lib/xmlenc.js
index bea6a5c..3477361 100644
--- a/node_modules/xml-encryption/lib/xmlenc.js
+++ b/node_modules/xml-encryption/lib/xmlenc.js
@@ -3,6 +3,7 @@ var xmldom  = require('@xmldom/xmldom');
 var xpath   = require('xpath');
 var utils   = require('./utils');
 var pki     = require('node-forge').pki;
+var forge   = require('node-forge');
 
 const insecureAlgorithms = [
   //https://www.w3.org/TR/xmlenc-core1/#rsav15note
@@ -237,7 +238,7 @@ function decryptKeyInfo(doc, options) {
     xpath.select("//*[local-name(.)='CipherValue']", keyInfo)[0];
 
   switch (keyEncryptionAlgorithm) {
-    case 'http://www.w3.org/2001/04/xmlenc#rsa-oaep-mgf1p':
+    case 'http://www.w3.org/2009/xmlenc11#rsa-oaep':
       return decryptKeyInfoWithScheme(encryptedKey, options, 'RSA-OAEP');
     case 'http://www.w3.org/2001/04/xmlenc#rsa-1_5':
       utils.warnInsecureAlgorithm(keyEncryptionAlgorithm, options.warnInsecureAlgorithm);
@@ -250,7 +251,8 @@ function decryptKeyInfo(doc, options) {
 function decryptKeyInfoWithScheme(encryptedKey, options, scheme) {
   var key = Buffer.from(encryptedKey.textContent, 'base64').toString('binary');
   var private_key = pki.privateKeyFromPem(options.key);
-  var decrypted = private_key.decrypt(key, scheme);
+  var decrypted = private_key.decrypt(key, scheme, {
+  md: forge.md.sha256.create()});
   return Buffer.from(decrypted, 'binary');
 }
 
