diff --git a/node_modules/xml-crypto/lib/signed-xml.js b/node_modules/xml-crypto/lib/signed-xml.js
index 7b23694..35ff4b8 100644
--- a/node_modules/xml-crypto/lib/signed-xml.js
+++ b/node_modules/xml-crypto/lib/signed-xml.js
@@ -122,10 +122,14 @@ function RSASHA256() {
   * Sign the given string using the given key
   *
   */
-  this.getSignature = function(signedInfo, signingKey, callback) {
+  this.getSignature = function (signedInfo, signingKey, callback) {
     var signer = crypto.createSign("RSA-SHA256")
     signer.update(signedInfo)
-    var res = signer.sign(signingKey, 'base64')
+    var res = signer.sign({
+      key: signingKey,
+      padding: crypto.constants.RSA_PKCS1_PSS_PADDING,
+      saltLength: crypto.constants.RSA_PSS_SALTLEN_DIGEST
+    }, 'base64')
     if (callback) callback(null, res)
     return res
   }
@@ -134,16 +138,20 @@ function RSASHA256() {
   * Verify the given signature of the given string using key
   *
   */
-  this.verifySignature = function(str, key, signatureValue, callback) {
+  this.verifySignature = function (str, key, signatureValue, callback) {
     var verifier = crypto.createVerify("RSA-SHA256")
     verifier.update(str)
-    var res = verifier.verify(key, signatureValue, 'base64')
+    var res = verifier.verify({
+      key: key,
+      padding: crypto.constants.RSA_PKCS1_PSS_PADDING,
+      saltLength: crypto.constants.RSA_PSS_SALTLEN_DIGEST
+    }, signatureValue, 'base64')
     if (callback) callback(null, res)
     return res
   }
 
   this.getAlgorithmName = function() {
-    return "http://www.w3.org/2001/04/xmldsig-more#rsa-sha256"
+    return "http://www.w3.org/2007/05/xmldsig-more#sha256-rsa-MGF1"
   }
 
 }
@@ -329,7 +337,7 @@ SignedXml.HashAlgorithms = {
 
 SignedXml.SignatureAlgorithms = {
   'http://www.w3.org/2000/09/xmldsig#rsa-sha1': RSASHA1,
-  'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256': RSASHA256,
+  'http://www.w3.org/2007/05/xmldsig-more#sha256-rsa-MGF1': RSASHA256,
   'http://www.w3.org/2001/04/xmldsig-more#rsa-sha512': RSASHA512,
   // Disabled by default due to key confusion concerns.
   // 'http://www.w3.org/2000/09/xmldsig#hmac-sha1': HMACSHA1
